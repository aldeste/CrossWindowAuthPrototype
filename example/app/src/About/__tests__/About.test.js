import React from "react";
import renderer from "react-test-renderer";
import About from "../About";

jest.mock("../../Tags/Tags");

const PersonInfo = {
  name: "Luke Skywalker",
  birthYear: "19BBY",
  eyeColor: "blue",
  hairColor: "blond",
  gender: "male",
  homeworld: {
    name: "Tatooine",
    residentConnection: {
      edges: [
        {
          node: {
            name: "Luke Skywalker"
          }
        },
        {
          node: {
            name: "C-3PO"
          }
        },
        {
          node: {
            name: "Darth Vader"
          }
        },
        {
          node: {
            name: "Owen Lars"
          }
        },
        {
          node: {
            name: "Beru Whitesun lars"
          }
        },
        {
          node: {
            name: "R5-D4"
          }
        },
        {
          node: {
            name: "Biggs Darklighter"
          }
        },
        {
          node: {
            name: "Anakin Skywalker"
          }
        },
        {
          node: {
            name: "Shmi Skywalker"
          }
        },
        {
          node: {
            name: "Cliegg Lars"
          }
        }
      ]
    }
  }
};

const sleep = ms =>
  new Promise(res => {
    setTimeout(res, ms);
  });

describe("About component", () => {
  global.fetch = () =>
    new Promise(resolve =>
      resolve({
        json: () => ({
          data: {
            viewer: PersonInfo,
            person: PersonInfo
          }
        })
      })
    );
  it("Starts att null", () => {
    expect(renderer.create(<About />).toJSON()).toBe(null);
  });

  it("Renders correctly on fetch", async () => {
    const component = renderer.create(<About />);
    await sleep(1);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
