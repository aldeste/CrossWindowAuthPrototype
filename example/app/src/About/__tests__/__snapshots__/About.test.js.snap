// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Renders correctly on fetch 1`] = `
<div
  className="View__View-jelnfZ bnYeJt"
>
  <h2
    className="TitleH2__TitleH2-dymhMn jBxvTF"
  >
    Luke Skywalker
  </h2>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    Born: 
    19BBY
  </p>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    Your hair is 
    blond
  </p>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    Your eyes are 
    blue
  </p>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    Your gender is 
    male
  </p>
  <h1
    className="TitleH3__TitleH3-jUefWF bXBjdj"
  >
    Your homeworld
  </h1>
  <div
    className="View__View-jelnfZ bnYeJt"
  >
    <p
      className="Text__Text-jpiXUO hvuOVR"
    >
      You're from 
      Tatooine
      ! Other people are also from there! Maybe you know them?
    </p>
    <p
      className="Text__Text-jpiXUO hvuOVR"
    >
      C-3PO, Darth Vader, Owen Lars, Beru Whitesun lars, R5-D4, Biggs Darklighter, Anakin Skywalker, Shmi Skywalker, Cliegg Lars
    </p>
  </div>
</div>
`;
